### ISSUE TITLE
Analytics [Unhandled promise rejection: Error: No credentials, applicationId or region]

### ISSUE THREAD

**Issue by kdchu** [aws-amplify/amplify-js#7053]
**Describe the bug**
I am attempting to add analytics to my application but it seems like the config is not being recognized in my awsconfig. The analytics information is there including the appId as well as region but the error still persists and no information is shown in Pinpoint (not even usage as others have mentioned). My auth allows for unauthenticated users as well, but I am also authenticated before performing analytics.record("example"). I have tried to completely remove amplify and reinit the project from scratch but nothing changed. However starting from a fresh project I was able to record and use analytics without an issue. I have also tried manually configuring Analytics with both Analytics.configure() as well as extending my
Amplfy.configure({
   ...awsconfig,
   Analytics: {
   }
});

**To Reproduce**

1. Create new expo project
2. Add the amplify CLI and aws-amplify
3. amplify add auth and analytics
4. attempt analytics.record("event")

**Expected behavior**
I expect to see some logged events in Pinpoint. As stated before I was able to do this with a brand new project but not with my current one.

**Code Snippet**
const awsconfig = {
    "aws_project_region": "xx-xxxx-x",
    "aws_cognito_identity_pool_id": "xxx",
    "aws_cognito_region": "xxx",
    "aws_user_pools_id": "xxx",
    "aws_user_pools_web_client_id": "xxx",
    "oauth": {
        "domain": "xxx.amazoncognito.com"
    },
    "aws_appsync_graphqlEndpoint": "xxx",
    "aws_appsync_region": "xxx",
    "aws_appsync_authenticationType": "xxx",
    "aws_appsync_apiKey": "xxx",
    "aws_mobile_analytics_app_id": "xxx",
    "aws_mobile_analytics_app_region": "xxx"
};

and within my App.js I run Amplify.configure(awsconfig);

**What is Configured?**
If applicable, please provide what is configured for Amplify CLI:
* Which steps did you follow via Amplify CLI when configuring your resources.
* Which resources do you have configured?
 
I currently have Auth, API, and Analytics configured. The only one that has given me any warnings is the Analytics. I used the CLI to add analytics and api. I imported auth using amplify import auth as per the recent release since I had cognito set up.

**Comment by amhinson** [comment]
Does your steps for reproduction actually produce the error you are seeing? I see that you also mention that you are not able to reproduce this error with a new project.

**Comment by kdchu** [comment]
@amhinson I am not completely sure what went wrong so I am unsure how to reproduce it without providing a public repo. I was able to fix the issue by ultimately creating a new project and bringing the old components in. I was able to import the same Auth and used the same configuration settings for everything from my IAM user to my API. My only thought is that I was missing something in the analytics module or an old analytics config was being recognized somewhere in the project. For now it seems to be resolved but I will keep looking at the issue in case it happens again in the future. Thanks for taking a look though.
